notebooks/data_table_transformer/main.py

```diff
@@ -0,0 +1,125 @@
+os.environ['WORKSPACE_BUCKET'] = "gs://fc-9169fcd1-92ce-4d60-9d2d-d19fd326ff10"
+os.environ['GOOGLE_PROJECT'] = "firecloud-cgl"
+
+with herzog.Cell("markdown"):
+    """
+    ### Use  Pandas `pivot` command  to convert [AnVIL](https://anvilproject.org) table from  long to wide
+    - `index_column` long format column containing the unique keys of the wide formatted table rows
+    - `header_column` long format column containing the new column headers of the wide formatted table
+    - `value_columns` long format columns to carry over into the wide formatted table (These may be muxed with the
+       `header_column` names, e.g. "cram" -> "file_type-cram")
+
+    While this notebook is focused on the AnVIL data model, it can work with any Terra data table with appropriate
+    choices for `index_column`, `header_column`, and `value_columns`.
+    """
+
+with herzog.Cell("markdown"):
+    """
+    Install the newest version of [terra-notebook-utils](https://github.com/DataBiosphere/terra-notebook-utils)
+    and [terra-pandas](https://github.com/xbrianh/terra-pandas)
+    """
+
+with herzog.Cell("python"):
+    #%pip install --upgrade --no-cache-dir terra-notebook-utils
+    #%pip install --upgrade --no-cache-dir terra-pandas
+    pass
+
+with herzog.Cell("markdown"):
+    """
+    Import Pandas interface functions from [terra-pandas](https://github.com/xbrianh/terra-pandas)
+    """
+
+with herzog.Cell("python"):
+    from terra_pandas import table_to_dataframe, dataframe_to_table, long_to_wide
+
+with herzog.Cell("markdown"):
+    """
+    Carry out the long to wide data table transformation, which involves several steps:
+      - Convert the Terra data table into a Pandas dataframe with `table_to_dataframe`
+      - Filter out rows that cannot be transformed into long format. In this example we are transforming the data table
+        into a wide format using sample id. Multisample files, such as jointly called VCFs, will not fit into the
+        transformed dataframe. There are also extra complexities of the AnVIL data model that need to be addressed,
+        as explained below.
+      - Convert the wide-formatted dataframe into a Terra data table.
+    """
+
+import json
+import pandas
+from uuid import uuid4
+from random import choice, randint
+from terra_notebook_utils import table
+
+NUMBER_OF_WIDE_ROWS = 20
+
+long_table = "test-long"
+wide_table = "test-wide"
+
+table.delete(long_table)
+table.delete(wide_table)
+
+lines = list()
+for _ in range(NUMBER_OF_WIDE_ROWS):
+    sample_id = f"{uuid4()}"
+    sample_obj = dict(foo=f"{uuid4()}", entityName=sample_id)
+    for fmt in ["cram", "crai", "vcf"]:
+        lines.append({'pfb:sample': sample_obj,
+                      'pfb:data_format': fmt,
+                      'pfb:object_id': f"{uuid4()}",
+                      'pfb:file_size': randint(1024, 1024 ** 3),
+                      'pfb:file_name': f"{uuid4()}"})
+dataframe_to_table("test-long", pandas.DataFrame(lines))
```

Comment:
Consider a comment that this section is purely for testing.

Reply:
Will add.